src/transports/mta/mta_transport.php

```diff
@@ -68,7 +68,7 @@ public function send( ezcMail $mail )
         $additionalParameters = "";
         if ( isset( $mail->returnPath ) )
         {
-            $sanitized = preg_replace( '([^' . ezcMail::RETURN_PATH_CHARS . '])', '', $mail->returnPath->email );
+            $sanitized = filter_var( $mail->returnPath->email, FILTER_SANITIZE_EMAIL );
```

Comment:
OK, so re-stating how I understand the issue -- the problem here is that upstream PHP's `mail()` is not compatible with some return-path addresses. The importance of this depends on the use-case/user/deployment:

* In some use-cases, having a precise return-path doesn't matter. For example, if an application sends error-logs out over email, then it probably doesn't track bounces. Moreover, it should try as hard as possible to send the error-log -- even if that means altering the return-path.
* In some use-cases, having a precise return-path is important. For example, a newsletter app may use [VERP](https://en.wikipedia.org/wiki/Variable_envelope_return_path) to track bounces and to remove defunct subscribers. Altering the return-path means that bounces won't get back to the right place. (The delivery-stats will incorrectly claim 100% delivery -- and it won't remove defunct subscribers.)

Perhaps a warning could balance these cases? e.g.

```suggestion
            $sanitized = preg_replace( '([^' . ezcMail::RETURN_PATH_CHARS . '])', '', $mail->returnPath->email );
            if ($sanitized !== $mail->returnPath->email)
            {
                error_log(sprintf('PHP mail() has limited support for return-paths. The return-path "%s" cannot be used. For proper return handling, configure a different transport or a different return-path.', $mail->returnPath->email));
            }
```

This way, if downstream has an invalid configuration, it will still make a best effort to deliver (*good for the logger use-case*) - but it will also make some noise so that the invalid configuration can be addressed (*good for the newsletter use-case*).

Reply:
I don't think the VERP case is particularly an issue, as the example in the RFC uses only `+`, `-`, and `=` in the Return Path, which is also what I mostly see happening in the real world™. So these wouldn't be sanitised out. I haven't quite seen a real life example case of for which return paths the original sanitiser wasn't mangling the return path. If you have examples, I'm curious to see.

In general, I am not keen on using `error_log` myself, and would rather see an exception thrown. I don't think that's much of a problem as I think for most cases, the original sanitiser wouldn't have stripped out characters for nearly all use cases.